3d adventure

Pewweper opened the pull request:
I added the prototype of this game that features:
-playable character that looks at the mouse position
-NPC that you can interact with
-Dialogue which has options.

Pewweper committed:
added initial stuff

Pewweper committed:
dialogue box + talking options

Added a placeholder mesh, and more importantly, you can interact with it.

Pewweper committed:
Merge branch 'main' into 3d_adventure

snoweuph reviewed (changes requested):
Dirty Commit history:
![image](https://github.com/user-attachments/assets/2355b4a3-ca33-4762-a26f-b617a9c902fd)

Not following branch naming.

Also, a lot of test code in this.
Test code shouldn't really get into the main code.


How about we get into a call quickly, so we can get some code rules down and basic setup for the project?

Pewweper committed:
added following units

they follow you in a snake-like formation.

Pewweper committed:
new follower scene

now in triangular formations

Pewweper committed:
camera work

use [, ], and \ for controls

Pewweper committed:
new meshes + updated ally logic

also added base thing

Pewweper committed:
Merge branch 'main' into 3d_adventure

Pewweper committed:
scrap collection mechanic added

Pewweper committed:
added damaging and being damaged

Pewweper committed:
Overworld + ally rework + dungeon

That's pretty much it

Pewweper committed:
enemies added

DragonfallGames committed:
A few minor changes

I just changed a few things, mostly setting any scene changes to be called deferred, and small camera adjustments.

DragonfallGames committed:
Small Camera Improvements

I managed to get the camera onto a spring arm so that it wont clip through walls as much.

DragonfallGames committed:
Improved player walking look direction

The player will now look in the direction they are walking unless attacking in which case it reverts to the previous behavior of facing the direction of the mouse.

DragonfallGames committed:
player input not rotated - fixed

The player input wasnt rotating correctly with the new camera system, I just realized this and fixed it.

Pewweper committed:
survival mode

Pewweper committed:
working survival mode

and also added gravity for enemy

Pewweper committed:
added_enemies

Pewweper committed:
added shooter into the game

Pewweper committed:
added instruction for shooter

DragonfallGames committed:
WIP Unit Throwing

I changed up the units - we may be leaving behind the follower system but I kept some of that in, either way I added the ability to manifest units that are thrown, using the combat unit throwing type.

Pewweper committed:
made player jump higher

DragonfallGames committed:
Added Unit Throwing Types

I added functionality for each unit types unique throwing style

Pewweper committed:
first level + fixed allies

Pewweper committed:
level_1 update + enemy_logic + ally_collecting

level_1 update + enemy_logic + ally_collecting

enemies now return to their position
level_1 has an exit portal and a pillar to climb on, along with the bridge

units can now be collected, by a press of a button.

Pewweper committed:
minor unit logic fixes

Pewweper committed:
level 1 complete(kinda)

Pewweper committed:
Level 2 also workshop added

And the enemy look_at() error is fixed that appears when they come back at their position

Pewweper committed:
added blueprints

YOU CAN BUILD STUFF NOW, YAY
Added defence turret
Still forgot to fix level 1's spawning point

DragonfallGames committed:
Reverted Unit Broken Functionality

I reverted the units back to a semi-functional state.

DragonfallGames committed:
Added Unit Recall Mechanics

Added mechanics for Units to be recalled when the e key is pressed.

DragonfallGames committed:
Unit Attack Logic

I made it so Units will properly go into attack mode and attack hostiles in their personal radius, rather than relying on the player being near enemies.

DragonfallGames committed:
quick clean

I submitted the last commit with a bit of redundant code and links since I used a different method and removed it.

Pewweper committed:
surival mode update

Pewweper committed:
added level 3 AND an a new addon

Pewweper committed:
level 3 update

Pewweper committed:
multiplayer testing

multiplayplay

Pewweper committed:
level 3 and multiplayer minor update

Pewweper committed:
multiplayer fixes: part 2

part 3 coming soon!!!

DragonfallGames committed:
Unit Scrap Collection

I added unit scrap collection where they will automatically collect scrap until all of the scrap is gone.

Pewweper committed:
player camera and ui fix, new boss added

New boss that shoots and jumps around like he's on crack or something

Fixed UI not appearing if it's not multiplayer on the player, and fixed camera on levels.

Pewweper committed:
multiplayer fixes

Players now respawn instead of changing the scene to "Overworld"

Pewweper committed:
shielder enemy and multiplayer stuff

Pewweper committed:
new mining rig building

and also added music slider
press M to mute/unmute music

Pewweper committed:
new waves and new medic enemy

New medic enemy that follows enemies and heals them.
Waves went from 10 -> 16

Pewweper committed:
more waves

wave amount 16(15) -> 20
also added wave 11
Yeah, the wave 11 was missing for some reason LOL

Pewweper committed:
heal_func for enemies, new upgrade, pausing.

The new upgrade is a shoulder gun
Gave ability to pause for the player.
The enemy medic healing certain enemy units would give an error that they do not have "heal_func()", but now they do.

Pewweper committed:
unit throw limit

Pewweper committed:
small fixes, new mode on survival

Fixed test boss having 0 damage
Fixed unit not being properly collected in multiplayer:i.e. collected, but the actual number of units not going up
Fixed resource piles being in different places
Buffed shouldergun: Shoots twice as faster (0.4s) and it's bullets can penetrate through enemies
New mode on survival, where enemies will attack from 4 positions at once

Pewweper committed:
adding "endless" to survival mode

The waves can now be generated, making the "endless survival mode" actually endless.

Pewweper committed:
returned sound to music

DragonfallGames committed:
Refactor Progress Start

I have begun refactoring, for now simply setting up the general entity script to have most of the base functions we will need. It is not yet integrated.

Pewweper committed:
multiple multiplayer fixes

Pewweper committed:
multiplayer fixes 1

Pewweper committed:
multiplayer fixes 2

Pewweper committed:
multiplayer fixes 3

Pewweper committed:
multiplayer fixes 4

DragonfallGames committed:
Refactoring Progress

This is the start of implementing refactoring, but there is a lot more to do, likely wont be done this deadline. This update may have bugs.

DragonfallGames committed:
Merge branch '3d_adventure' of https://github.com/RedotOpenGame/game into 3d_adventure

DragonfallGames committed:
Minor Quickfix

A small quickfix for the last push so the game doesnt crash on launch.

Pewweper committed:
Multiplayer conveniences

Added nicknames above player
Added nicknames above things that belong to player(units) or made by player (buildings)
Added in-game chat
Also fixed blueprint and module gun not working in multiplayer (for a second time)

Pewweper committed:
painted units and random names for multplr

Quadratus873 committed:
Add files via upload

A Prototype City

Pewweper committed:
mesh imported

Quadratus's mesh of "ProtoCity" was imported into the game.

Pewweper committed:
small changes, refactoring

Pewweper committed:
devtools at overworld, enemy fixes

Pewweper committed:
QoL changes

Enemies shoot when in certain range
dev tools(enemy spawner/killer)
Enemy fixes

Pewweper committed:
optimizations mostly regarding units

Pewweper committed:
added boombox

and made buildings drop scrap when dying (80% of original cost)

Pewweper committed:
mortar and balance change

Pewweper committed:
QoL stuff and bug fixes

Demolish buildings on Y, Ctrl+Z to call all units, regardless of what they're doing, and chat bugfix, where it would scroll all the way up everytime someone sends a message(you had to manually scroll down for that)

And bug fixes.

Pewweper committed:
new music for survival

Pewweper committed:
new enemy tier and wave autoskip

Pewweper committed:
unit collection showing

Pewweper committed:
new ui + new font

Pewweper committed:
blueprint placement rework

Pewweper committed:
small fixes

blueprint can be now rotated by pressing R
Player can place workshop/blueprint depensing on mouse position(also constrained within range of 10 meters)

Pewweper committed:
added unit factory

Pewweper committed:
new attachment

Pewweper committed:
new turret

Added sniper turret, and also non-collector units do not collect scrap anymore(finally)

Pewweper committed:
more feature bloat

Added ability to block(unable to attack when doing that + 0.1x damage when blocking), press Q
Added ability to lock-on to enemies, press G in order to look at the nearest enemy
Added labels onto demolishing, collection and lock-on to show what buttons must be pressed in order to activate.

Pewweper committed:
level 3 update

Added platformer and "third wall"

Pewweper committed:
new level + tutorial finished

Pewweper committed:
first level done

Pewweper committed:
Level 2 complete

Pewweper committed:
LEVEL 3

the maze, not very big though.

Pewweper committed:
level 4 W.I.P.

Pewweper committed:
new room in level 4
